src/main/java/frc/robot/subsystems/SwerveDriveSubsystem.java

```diff
@@ -137,14 +119,27 @@ public void move(double xSpeed, double ySpeed, double rotationSpeed, boolean isF
     m_ySpeed = ySpeed;
     m_rotationSpeed = rotationSpeed;
     m_isFieldRelative = isFieldRelative;
+    m_isTurning = rotationSpeed != 0;
   }
 
   /**
-   * Resets the gyro.
-   * Aka, sets the current heading of the robot to zero.
-   * TODO make sure this also updates odometry, <i>if needed</i>.
+   * Resets the gyro. Note that whatever command calls this method <b>will require
+   * the subsystem</b>. So no other commands can run on the subsystem while the
+   * gyro is being reset, including any drive control commands.
```

Comment:
does this still apply for an instant command?

Reply:
Good point.  If you look at [the docs for that InstantCommand constructor](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/InstantCommand.html#%3Cinit%3E(java.lang.Runnable,edu.wpi.first.wpilibj2.command.Subsystem...)), the subsystem that I pass in as the second parameter is the subsystem that the command requires.